lib/operations/find_and_modify.js

```diff
@@ -22,17 +22,16 @@ class FindOneAndDeleteOperation extends OperationBase {
   execute(callback) {
     const coll = this.collection;
     const query = this.query;
-    let sort = this.sort;
+    const sort = formattedOrderClause(this.sort);
     const doc = this.doc;
-    const options = this.options;
+    const options = Object.assign({}, this.options);
```

Comment:
I don't think we need to do this if we make sure that nothing is directly modifying `options` below. We already create a `finalOptions` which is a clone of this one (so we're doing this all twice), and it looks like the one case where it modifies `options` is to add a `checkKeys: false`, which could be moved to `finalOptions`.

Reply:
turns out these are all already clones in their parent operations, so in some cases we're doing an `Object.assign` up to three times